src/commands/build.rs

```diff
@@ -1,16 +1,84 @@
 pub async fn build(config: &fpm::Config) -> fpm::Result<()> {
     tokio::fs::create_dir_all(config.build_dir()).await?;
 
-    let (documents, config) = fpm::get_documents_with_config(config).await?;
+    let translation_snapshots = if let Some(ref original) = config.package.translation_of.as_ref() {
+        let translation_config = config.read_translation().await?;
+        let documents = config
+            .get_translation_documents()
+            .await?
+            .iter()
+            .map(|v| {
+                (
+                    v.get_id()
+                        .replace(&format!(".packages/{}/", original.name), ""),
+                    v.to_owned(),
+                )
+            })
+            .collect::<Vec<(String, fpm::File)>>();
+
+        std::env::set_current_dir(
+            &config
+                .original_directory
+                .join(".packages")
+                .join(original.name.as_str()),
+        )?;
+        build_(
+            &translation_config,
+            &config.original_directory,
+            documents,
+            true,
+        )
+        .await?;
+
+        // is it needed??
+        fpm::utils::copy_dir_all(".packages", config.original_directory.join(".packages")).await?;
+        std::env::set_current_dir(&config.original_directory)?;
+
+        Some(fpm::snapshot::get_latest_snapshots(&config.original_path()?).await?)
+    } else {
+        None
+    };
+
+    let documents = {
+        let mut documents = fpm::get_root_documents(config)
+            .await?
+            .into_iter()
+            .map(|v| (v.get_id(), v))
+            .collect::<Vec<(String, fpm::File)>>();
+        if let Some(translation_snapshots) = translation_snapshots {
+            documents = documents
+                .into_iter()
+                .filter(|(_, x)| translation_snapshots.contains_key(x.get_id().as_str()))
+                .collect::<Vec<(String, fpm::File)>>();
+        }
+        documents
+    };
```

Comment:
Write a function with descriptive name and put this block inside the function, or write comments to explain what is happening. These are extremely complex code, don't write code like this with zero explanation etc.